### ISSUE TITLE
ate function produces errors and NaN values when using method = "AIPTW"

### ISSUE THREAD

**Issue by RobinDenz1** [tagteam/riskRegression#18]
In some cases the `ate` function produces NaN values or returns an error code in `G.jump$survival * S.jump$survival`. I sadly can't share the real data here, but I managed to simulate data and reproduce the problem.

The simulated data is inside the attached file: [Sim_Dat.zip](https://github.com/tagteam/riskRegression/files/5457317/Sim_Dat.zip).

Since it is simulated, I know the true data generation models. Both the treatment and the outcome model are correctly specified. I used the following code:

```
library(riskRegression)
library(survival)

sim_dat <- read.csv("Sim_Dat.csv", sep="\t")
sim_dat$group <- factor(sim_dat$group)

# correctly specified outcome model
outc_model <- coxph(Surv(time, cause) ~ x1 + x2 + x4 + x5 + group,
                    data=sim_dat, x=T)
# correctly specified treatment model
treat_model <- glm(group ~ x2 + x3 + x5 + x6, data=sim_dat,
                   family="binomial")
# censoring is random, so this model is technically wrong
cens_model <- coxph(Surv(time, cause==0) ~ group, data=sim_dat, x=T)

# for times < 1 it returns errors
curve <- riskRegression::ate(event=outc_model, 
                             treatment=treat_model,
                             censor=cens_model,
                             data=sim_dat,
                             times=0.9,
                             se=FALSE, 
                             estimator="AIPTW")
# for times > 1 it returns NaN only
curve <- riskRegression::ate(event=outc_model, 
                             treatment=treat_model,
                             censor=cens_model,
                             data=sim_dat,
                             times=1.2,
                             se=FALSE, 
                             estimator="AIPTW")
```
For values smaller than 1, an error is produced. For values bigger than 1, the function simply produces NaN values without any warnings or errors. The IPTW and Gformula methods work fine, this error occurs only for `method="AIPTW"`.

Do you have any ideas what may cause this issue?

Here is the output to the obligatory `sessionInfo()` call:
```
R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252    LC_MONETARY=German_Germany.1252
[4] LC_NUMERIC=C                    LC_TIME=German_Germany.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] survival_3.1-12           riskRegression_2020.02.05

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5          mvtnorm_1.1-1       lattice_0.20-41     zoo_1.8-8           png_0.1-7          
 [6] assertthat_0.2.1    digest_0.6.25       foreach_1.5.0       R6_2.4.1            backports_1.1.10   
[11] MatrixModels_0.4-1  ggplot2_3.3.2       pillar_1.4.6        rlang_0.4.7         multcomp_1.4-14    
[16] rstudioapi_0.11     data.table_1.13.0   SparseM_1.78        rpart_4.1-15        Matrix_1.2-18      
[21] checkmate_2.0.0     splines_4.0.2       stringr_1.4.0       foreign_0.8-80      htmlwidgets_1.5.1  
[26] munsell_0.5.0       numDeriv_2016.8-1.1 compiler_4.0.2      xfun_0.17           pkgconfig_2.0.3    
[31] base64enc_0.1-3     htmltools_0.5.0     nnet_7.3-14         tidyselect_1.1.0    tibble_3.0.3       
[36] gridExtra_2.3       htmlTable_2.1.0     prodlim_2019.11.13  Hmisc_4.4-1         rms_6.0-1          
[41] codetools_0.2-16    matrixStats_0.56.0  fansi_0.4.1         crayon_1.3.4        dplyr_1.0.2        
[46] conquer_1.0.2       timereg_1.9.8       MASS_7.3-51.6       grid_4.0.2          nlme_3.1-148       
[51] polspline_1.1.19    gtable_0.3.0        lifecycle_0.2.0     magrittr_1.5        scales_1.1.1       
[56] cli_2.0.2           stringi_1.5.3       latticeExtra_0.6-29 ellipsis_0.3.1      generics_0.0.2     
[61] vctrs_0.3.4         sandwich_3.0-0      TH.data_1.0-10      Formula_1.2-3       lava_1.6.8         
[66] RColorBrewer_1.1-2  iterators_1.0.12    tools_4.0.2         cmprsk_2.2-10       glue_1.4.2         
[71] purrr_0.3.4         jpeg_0.1-8.1        colorspace_1.4-1    cluster_2.1.0       knitr_1.30         
[76] quantreg_5.67
```

**Comment by bozenne** [comment]
Thanks for the nice reproducible example. It was very helpful.
I have pushed a new version of the riskRegression package (2020-10-29) in the branch master that should fix the two issues.
-One was that, in your example, you have some individuals for which the predicted survival at time 0.9 is very close to 0.
Since some of the formula involve the at risk indicator divided by the survival  that was leading to 0/0 ie NA.
So now I have implemented that in a better such that, loosely speaking, 0/0 = 0 since the at risk indicator is exactly 0.

**Comment by RobinDenz1** [comment]
First of all, thank you for the fast reply and explanation. The way you are mantaining your R-Package is exemplary!

I installed your newly pushed version from github just now and tried to run the example I posted above. Sadly it still doesn't work. Now it returns a different error message: `Error in (function (object.event, object.treatment, object.censor, mydata,  : 
  Objekt 'index.beforeEvent' not found` (no matter what value for `times` is supplied).
